File: src/app/shared/service/api/mark.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { MarkDTORequest } from '../../model/request/markDTORequest';

@Injectable({
    providedIn: 'root'
})

export class MarkService {
    constructor(
        private _httpClient: HttpClient
    ) { }

    findAll(key_word: string, page: number, size: number): Observable<any> {
        let queryParameters = new HttpParams();
        queryParameters = queryParameters.append('page', page);
        queryParameters = queryParameters.append('size', size);
        queryParameters = queryParameters.append('field', '');
        queryParameters = queryParameters.append('order', '');
        queryParameters = queryParameters.append('key_word', key_word);
        return this._httpClient.get<any>(environment.urlBase + '/marks/mark?' + queryParameters);
      }
    
      public save(mark: MarkDTORequest): Observable<any> {
        return this._httpClient.post<any>(environment.urlBase + '/marks/mark', mark);
      }
    
      public findById(id: number): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/marks/mark/' + id);
      }
    
      public update(mark: MarkDTORequest, id: number): Observable<any> {
        return this._httpClient.put<any>(environment.urlBase + '/marks/mark/' + id, mark);
      }
    
      public delete(id: number): Observable<any> {
        return this._httpClient.patch<any>(environment.urlBase + '/marks/mark/' + id,null);
      }
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/MarkService.java
package com.sys.botica.crce.pe.sys_botica.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.MarkDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.MarkDTORequest;

public interface MarkService {
	public Page<MarkDTO> findAll(String key_word,Pageable pageable);
	public HrefEntityDTO save(MarkDTORequest mark);
	public HrefEntityDTO update(MarkDTORequest mark,Integer id);
	public HrefEntityDTO delete(Integer id);
	public MarkDTO findById(Integer id);
}
